src/log_surgeon/finite_automata/RegexAST.hpp

```diff
@@ -892,11 +892,19 @@ template <typename NFAStateType>
 }
 
 template <typename NFAStateType>
-void RegexASTCapture<NFAStateType>::add_to_nfa(RegexNFA<NFAStateType>* nfa, NFAStateType* end_state)
-        const {
-    auto* state_with_positive_tagged_transition
-            = nfa->new_state_with_positive_tagged_transition(m_tag.get(), end_state);
-    m_group_regex_ast->add_to_nfa_with_negative_tags(nfa, state_with_positive_tagged_transition);
+void RegexASTCapture<NFAStateType>::add_to_nfa(
+        RegexNFA<NFAStateType>* nfa,
+        NFAStateType* dest_state
+) const {
+    // root --(pos_tagged_start_transition)--> capture_group_start_state -->
+    // [inner capture group NFA] --(neg_tagged_transition)--> neg_state -->
+    // state_with_positive_tagged_end_transition --(pos_tagged_end_transition)--> end_state
+    auto [start_state, end_state] = nfa->new_capture_group_start_states(m_tag.get(), dest_state);
```

Comment:
It would be a bit verbose, but I think we can make it more readable by having a more meaningful name, and sync the name in the comments above:
```suggestion
    auto [positive_tagged_transition_start, positive_tagged_transition_end]
            = nfa->new_positive_tagged_transition_states(m_tag.get(), dest_state);
```

Reply:
Maybe we stick with `capture_group_start_state` and `capture_group_end_state` as implicitly these must have their respective transitions to be the start/end of the capture. And the diagram reinforces this idea for anyone unfamiliar with the NFA of a capture group.

Reply:
Sure. But for nfa's factory function, I think we should still call it positive_tagged_transition_states

Reply:
I'm following: https://google.github.io/styleguide/cppguide.html#General_Naming_Rules:
```
Generally speaking, descriptiveness should be proportional to the name's scope of visibility. For example, n may be a fine name within a 5-line function, but within the scope of a class, it's likely too vague.
```
Unless we have a specific guideline that overrules this.
```
template <typename NFAStateType>
auto RegexNFA<NFAStateType>::new_start_and_end_states_with_positively_tagged_transitions(
        Tag const* tag,
        NFAStateType const* dest_state
) -> std::pair<NFAStateType*, NFAStateType*> {
    auto* start_state = new_state();
    m_root->add_positive_tagged_start_transition(tag, start_state);

    auto* end_state = new_state_with_positive_tagged_end_transition(tag, dest_state);
    return {start_state, end_state};
}
```
Within its scope of visibility, there is no ambiguity as to which start and end states these reference.

Reply:
1. I'm not suggesting we should change `new_start_and_end_states_with_positively_tagged_transitions`. `start` and `end` states there are fine
2. The reason I commented on `auto [positive_tagged_transition_start, positive_tagged_transition_end]` is for consistency with the diagram in the comment

Reply:
Got it, updated the diagram now that the variable names are fixed.